add.go

```diff
@@ -434,7 +434,7 @@ func (b *Builder) Add(destination string, extract bool, options AddAndCopyOption
 			destination = tmpDestination
 		}
 	}
-	destMustBeDirectory := (len(sources) > 1) || strings.HasSuffix(destination, string(os.PathSeparator)) || destination == b.WorkDir()
+	destMustBeDirectory := (len(sources) > 1) || strings.HasSuffix(destination, string(os.PathSeparator)) || path.Base(destination) == "." || destination == b.WorkDir()
```

Comment:
AFAICS 
```go
tmpDestination := filepath.Join(string(os.PathSeparator)+b.WorkDir(), destination)
```
above turns `filepath.Join("/foo", "/bar/.")` into `"/foo/bar"`, erasing the `/.` suffix, so I think this might not be exactly right. (I’m not familiar with, and didn’t now review, the whole call tree path, to see whether there is any better place to make this decision.)

Reply:
Okay, splitting that up so that we look at the passed-in destination value when using it to guess the intent.